common/src/main/java/com/wynntils/features/user/CombatXPGainMessageFeature.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright © Wynntils 2022.
+ * This file is released under AGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.features.user;
+
+import com.wynntils.core.config.Config;
+import com.wynntils.core.features.UserFeature;
+import com.wynntils.core.features.properties.FeatureInfo;
+import com.wynntils.core.notifications.NotificationManager;
+import com.wynntils.mc.event.ClientTickEvent;
+import com.wynntils.wynn.model.CharacterManager;
+import com.wynntils.wynn.utils.WynnUtils;
+import net.minecraftforge.eventbus.api.SubscribeEvent;
+
+@FeatureInfo
+public class CombatXPGainMessageFeature extends UserFeature {
+
+    @Config
+    public boolean getCombatXPGainMessages = true;
+
+    private static float newTickXP = 0;
+    private static float lastTickXP = 0;
+    private static int trackedPercentage = 0;
+
+    @SubscribeEvent
+    public void onTick(ClientTickEvent.End event) {
+        if (!WynnUtils.onWorld() || !getCombatXPGainMessages) {
+            return;
+        }
+        CharacterManager.CharacterInfo data = WynnUtils.getCharacterInfo();
+
+        newTickXP = data.getCurrentXp();
+
+        if (newTickXP == lastTickXP) {
+            return;
+        }
+
+        int neededXP = data.getXpPointsNeededToLevelUp();
+        if (lastTickXP != 0) {
```

Comment:
I am not sure if defaulting to 0 is beneificial, does this work nicely with level ups?

Reply:
The only time when experience is 0 is at the very start of a session, when you join a world. Otherwise, it will be a positive integer by some degree. There might be an edge case regarding that where you could get exactly 0 points, but that'll smooth over by the next second. Not sure what the best way to handle that edge case is, or even it's important. 

I'll test it in-game with a levelup and see what comes of it to ensure nothing's breaking.

Reply:
But theoritically you could stay on 0 xp. I think. So maybe, use a different magic number for the edge case? -1?

Reply:
If you "stay on" 0 XP then you get filtered by this above check:

```java
if (newTickXP == lastTickXP) return;
```

or do I misunderstand? This code only runs whenever there is an actual change in the XP from the last time we ran.